src/parser.rs

```diff
@@ -114,14 +114,21 @@ fn parse_err(args: &[u8]) -> ControlOp {
     ControlOp::Err(err_description.to_string())
 }
 
-pub(crate) fn expect_info(reader: &mut BufReader<TcpStream>) -> io::Result<ServerInfo> {
-    let op = parse_control_op(reader)?;
+pub(crate) fn expect_info(reader: &mut TcpStream) -> io::Result<ServerInfo> {
+    let mut buf = Vec::with_capacity(512);
 
-    if let ControlOp::Info(info) = op {
-        Ok(info)
-    } else {
-        Err(Error::new(ErrorKind::Other, "INFO proto not found"))
+    while !buf.ends_with(b"\r\n") {
```

Comment:
I see now what you did here. This might cause a hiccup in initial connect speed which we should track. Maybe add a TODO(spacejam) comment to revisit and see if we can come up with something better. We have this in alot of other clients of course and I don't think we read byte by byte. Might be worth looking quickly at Go client.